### ISSUE TITLE
Alert a11y

### ISSUE THREAD

**Issue by Heydon** [iTwin/stratakit#1220]
## 1. Downgrade live region role

Despite the name, MUI alerts are really just status messages (a "success" example is included). However, they use the `role="alert"` live region semantics. 

This should be downgraded to `role="status"` and paired with `aria-live="polite"` for legacy compatibility. 

## 2. Make Alerts discoverable

Alerts (de facto status messages) do not intrinsically belong to landmarks. Messages can be missed, so there needs to be a "message center" to check back.

This either needs to be addressed in code (by extending the markup) or as part of example implementations in the guidance. A labeled `<aside>` landmark, containing all messages, should suffice. A single live region can take successive messages. The `aria-relevant="additions"` attribution ensures only new messages are announced.

```html
<aside aria-label="Status messages">
    <div role="status" aria-relevant="additions">
        <!-- status message 1 -->
        <!-- status message 2 -->
    </div>
</aside>
```

## 3. Group alert contents

MUI Alerts (status messages) currently use their own `role="alert"` attribution. This is recommended against in **(2)**, which prescribes a discoverable "message center". 

Individual alerts are not always simple text strings. Some have structured content, including headings. To communicate which content belongs to which alert, the alert itself should, instead, have `role="group"`. See the code example in **(4)**.

## 4. Alert heading should use heading semantics or be hidden

The success example (https://supreme-barnacle-pl8jn8m.pages.github.io/mui/#alert) uses a faux heading (**Success**). If we believe Alerts should use headings, these should have heading semantics supplied.

If the would-be heading is deemed superfluous to the document outline, it's acceptable to make it unavailable through browsing (`aria-hidden="true"`) and instead use it to label the status message itself.

```html
<div aria-labelledby="status-message-1-label" role="group">
    <div id="status-message-1-label" aria-hidden="true">Success</div>
    <!-- content -->
</div>
```

## 5. Icons have no alternative text

There is no mechanism for indicating the type/level of alert (error, warning, success, etc). The simplest way would be to give the icons prefixing the content alternative text. For this to be available in screen readers, `aria-hidden` would have to be removed.

```html
<svg class="🥝Icon" role="img" aria-label="Success">
    <use href="/assets/status-success-2QalDbE4.svg#icon"></use>
</svg>
```
